libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -14,6 +17,12 @@ AP_Camera_Backend::AP_Camera_Backend(AP_Camera &frontend, AP_Camera_Params &para
     _instance(instance)
 {}
 
+// common initialisation for all backends
+void AP_Camera_Backend::init()
+{
+    hal.scheduler->register_io_process(FUNCTOR_BIND_MEMBER(&AP_Camera_Backend::calculate_poi, void));
```

Comment:
This is not creating a new thread; this is adding an IO process (which runs your code on a thread, but not its own thread).

Randy pointed out the thread-create API to you.

Reply:
Yeah done it in the new patch.